### PR TITLE
SNOW-564006 Refactor telemetry code to make it extensible for Snowpipe Streaming

### PR DESCRIPTION
**Pull request by sfc-gh-japatel**
Refactor code to make it extensible for adding telemetry for Snowpipe Streaming

- Moved all telemetry classes to a new package. 
- All member variables to private and added setter and getter wherever required. 
- Added javadocs for new classes and new public methods (Except setter and getters since they are self explanatory)
- No behavior change. 
- Changed interface method names to remove pipe from it. Can be used by channels too!

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/internal/telemetry/SnowflakeTelemetryPipeCreation.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import static com.snowflake.kafka.connector.internal.telemetry.TelemetryConstants.IS_REUSE_TABLE;
+import static com.snowflake.kafka.connector.internal.telemetry.TelemetryConstants.PIPE_NAME;
+import static com.snowflake.kafka.connector.internal.telemetry.TelemetryConstants.STAGE_NAME;
+import static com.snowflake.kafka.connector.internal.telemetry.TelemetryConstants.START_TIME;
+import static com.snowflake.kafka.connector.internal.telemetry.TelemetryConstants.TABLE_NAME;
+
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.node.ObjectNode;
+
+// This object is send only once when pipe starts
+// No concurrent modification is made on this object, thus no lock is required.
```
Comment (sfc-gh-tzhang) [inline src/main/java/com/snowflake/kafka/connector/internal/telemetry/SnowflakeTelemetryPipeCreation.java:16]: javadoc format

File: src/main/java/com/snowflake/kafka/connector/internal/telemetry/SnowflakeTelemetryPipeStatus.java [hunk 2]
```diff
@@ -338,4 +370,91 @@ private void registerPipeJMXMetrics(
...
+
+  public void addAndGetMemoryUsage(long memoryUsage) {
+    this.memoryUsage.addAndGet(memoryUsage);
+  }
+
+  public void resetMemoryUsage() {
+    this.memoryUsage.set(0l);
+  }
+
+  // --------------- For testing --------------- //
```
Comment (sfc-gh-tzhang) [inline src/main/java/com/snowflake/kafka/connector/internal/telemetry/SnowflakeTelemetryPipeStatus.java:432]: Should we add @VisibleForTesting for all the functions below?

**Review by sfc-gh-tzhang — APPROVED** [review]
LGTM, thanks for doing this!

File: src/main/java/com/snowflake/kafka/connector/internal/telemetry/SnowflakeTelemetryPipeCreation.java (hunk 1 above)
Comment (sfc-gh-japatel) [reply src/main/java/com/snowflake/kafka/connector/internal/telemetry/SnowflakeTelemetryPipeCreation.java:16]: Copy paste issue. Will fix.

File: src/main/java/com/snowflake/kafka/connector/internal/telemetry/SnowflakeTelemetryPipeStatus.java (hunk 2 above)
Comment (sfc-gh-japatel) [reply src/main/java/com/snowflake/kafka/connector/internal/telemetry/SnowflakeTelemetryPipeStatus.java:432]: Good idea. Will do!

**Pull request merged**